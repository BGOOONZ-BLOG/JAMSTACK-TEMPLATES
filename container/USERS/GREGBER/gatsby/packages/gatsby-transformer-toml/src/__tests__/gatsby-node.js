const { onCreateNode } = require(`../gatsby-node`)

describe(`Process TOML nodes correctly`, () => {
  const node = {
    id: `whatever`,
    parent: null,
    children: [],
    extension: `toml`,
    internal: {
      contentDigest: `whatever`,
    },
    name: `test`,
  }

  // Provide fake functions
  const loadNodeContent = node => Promise.resolve(node.content)

  it(`Correctly creates nodes from TOML test file`, async () => {
    // Unfortunately due to TOML limitations no JSON -> TOML converters exist,
    // which means that we are stuck with JS template literals.
    node.content = `
    [the]
    test_string = "You'll hate me after this - #"

    [the.hard]
    test_array = [ "] ", " # "]      # ]
    test_array2 = [ "Test #11 ]proved that", "Experiment #9 was a success" ]
    another_test_string = " Same thing, but with a string #"
    harder_test_string = " And when \\"'s are in the string, along with # \\""
    # Things will get harder

        [the.hard."bit#"]
        "what?" = "You don't think some user won't do that?"
        multi_line_array = [
            "]",
            # ] Oh yes I did
            ]
    `

    const createNode = jest.fn()
    const createParentChildLink = jest.fn()
    const actions = { createNode, createParentChildLink }
    const createNodeId = jest.fn()
    createNodeId.mockReturnValue(`uuid-from-gatsby`)
    const createContentDigest = jest.fn().mockReturnValue(`contentDigest`)

    await onCreateNode({
      node,
      loadNodeContent,
      actions,
      createNodeId,
      createContentDigest,
    }).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(createParentChildLink.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(1)
      expect(createParentChildLink).toHaveBeenCalledTimes(1)
      expect(createContentDigest).toHaveBeenCalledTimes(1)
    })
  })

  // Since TOML transformer doesn't generate sub-objects from arrays,
  // but directly uses the object, 'id' uniqueness tests between sub-objects
  // are omitted.
})
