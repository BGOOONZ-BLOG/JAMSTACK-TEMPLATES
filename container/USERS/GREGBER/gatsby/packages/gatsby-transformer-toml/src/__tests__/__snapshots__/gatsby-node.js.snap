// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process TOML nodes correctly Correctly creates nodes from TOML test file 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "",
      },
      "parent": "whatever",
      "the": Object {
        "hard": Object {
          "another_test_string": " Same thing, but with a string #",
          "bit#": Object {
            "multi_line_array": Array [
              "]",
            ],
            "what?": "You don't think some user won't do that?",
          },
          "harder_test_string": " And when \\"'s are in the string, along with # \\"",
          "test_array": Array [
            "] ",
            " # ",
          ],
          "test_array2": Array [
            "Test #11 ]proved that",
            "Experiment #9 was a success",
          ],
        },
        "test_string": "You'll hate me after this - #",
      },
    },
  ],
]
`;

exports[`Process TOML nodes correctly Correctly creates nodes from TOML test file 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "children": Array [],
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "",
        },
        "parent": "whatever",
        "the": Object {
          "hard": Object {
            "another_test_string": " Same thing, but with a string #",
            "bit#": Object {
              "multi_line_array": Array [
                "]",
              ],
              "what?": "You don't think some user won't do that?",
            },
            "harder_test_string": " And when \\"'s are in the string, along with # \\"",
            "test_array": Array [
              "] ",
              " # ",
            ],
            "test_array2": Array [
              "Test #11 ]proved that",
              "Experiment #9 was a success",
            ],
          },
          "test_string": "You'll hate me after this - #",
        },
      },
      "parent": Object {
        "children": Array [],
        "content": "
    [the]
    test_string = \\"You'll hate me after this - #\\"

    [the.hard]
    test_array = [ \\"] \\", \\" # \\"]      # ]
    test_array2 = [ \\"Test #11 ]proved that\\", \\"Experiment #9 was a success\\" ]
    another_test_string = \\" Same thing, but with a string #\\"
    harder_test_string = \\" And when /\\"'s are in the string, along with # /\\"\\"
    # Things will get harder

        [the.hard.\\"bit#\\"]
        \\"what?\\" = \\"You don't think some user won't do that?\\"
        multi_line_array = [
            \\"]\\",
            # ] Oh yes I did
            ]
    ",
        "extension": "toml",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
]
`;
